crates/core/c8y_api/src/smartrest/alarm.rs

```diff
@@ -16,11 +16,23 @@ pub fn serialize_alarm(alarm: ThinEdgeAlarm) -> Result<String, SmartRestSerializ
                 AlarmSeverity::Warning => 304,
             };
             let current_timestamp = OffsetDateTime::now_utc();
+
+            let text = if alarm_data.text.is_none() && alarm_data.message.is_none() {
+                alarm_data.text.clone().unwrap_or_default()
+            } else if alarm_data.text.is_none() && alarm_data.message.is_some() {
+                alarm_data
+                    .message
+                    .clone()
+                    .unwrap_or_else(|| alarm.name.clone())
+            } else {
+                alarm_data.text.unwrap_or_default()
+            };
```

Comment:
This code is duplicated and should be shared as method of `C8yCreateAlarm`.

Reply:
Reverted the changes, not necessary anymore.